config/initializers/rack/attack.rb

```diff
@@ -0,0 +1,7 @@
+class Rack::Attack
+  throttle("search_throttle", limit: 5, period: 1) do |request|
+    if request.path.starts_with?("/search/")
+      request.ip.to_s
```

Comment:
Given that we currently use [shielding](https://docs.fastly.com/en/guides/shielding) I'm not 100% sure the typical request IPs will work as expected.

I haven't fully looked into it, just raising the possibility. I suspect it probably works just fine like this as intended but I just want to raise the question.

One other thing.... We might want to also implement this on the edge. I like that we have it here in the app because because it doesn't 100% tie us into the CDN provider, but Fastly has a lot of features for this built in and everything passes through their servers before it hits us so any logic like this can also live there. Having redundancy for this kind of thing can only be a good thing.

All non-blocking comments provided there's no problems accessing the request IP like this in prod.

Reply:
Looks like you are correct, the IPs we get seem to be from Fastly rather than our actual end-users which makes throttling by them not possible here. I am looking into our options via Fastly for throttling IPs, https://www.fastly.com/blog/restricting-access-content

Reply:
Cool. Fastly is built for throttling, so tools-wise this will ultimately be a useful thing, throttling before things even hit our servers is the best case scenario anyway.

But now that we know for sure that this issue exists, we should probably still look into it to find the appropriate way to deal with IPs like this in general.

Reply:
Oh, and this is a good chance to additionally move Fastly config into our application logic in part of that migration.

Reply:
Thanks to some Honeybadger investigating there is a way to throttle IPs on our end, via HTTP_FASTLY_CLIENT_IP which is the IP that hits Fastly and then is stored apparently in an env variable in the request.